app/register/controller.js

```diff
@@ -1,41 +1,39 @@
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
+import { Changeset } from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import EditUserValidations from '../validations/edit-user';
 
 export default class RegisterController extends Controller {
   @service store;
   @service router;
 
-  @action
-  onLoginChange(event) {
-    this.model.username = event.target.value;
-  }
-
-  @action
-  onEmailChange(event) {
-    this.model.email = event.target.value;
+  user = this.store.createRecord('user');
+  constructor() {
+    super(...arguments);
+    this.registerChangeset = new Changeset(
+      this.user,
+      lookupValidator(EditUserValidations),
+      EditUserValidations
+    );
```

Comment:
Out of `constructor` and `function`s define class attributes like that just if it is simple thing like `string`, `number` etc, definitely not asynchronous things
```suggestion
  constructor() {
    super(...arguments);
    this.user = this.store.createRecord('user');
    this.registerChangeset = new Changeset(
      this.user,
      lookupValidator(EditUserValidations),
      EditUserValidations
    );
```